### ISSUE TITLE
On POST requests, req.body is undefined in 0.12.4 while in 0.12.3 it existed as an empty object

### ISSUE THREAD

**Issue by luislobo** [balderdashy/sails#3803]
**Sails version**: 0.12.4
**Node version**: 4.4.5
**NPM version**:  3.10.3
**Operating system**: Linux 4.4.0-31-generic #50-Ubuntu SMP Wed Jul 13 00:07:12 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

<hr/>
In Sails 0.12.4, on post requests, req.body is undefined, if no body is sent.
In Sails 0.12.3, on post requests, req.body is {}, if no body is sent.
Steps to reproduce.
Clone https://github.com/luislobo/sails-body-error
start sails
Issue:
```bash
curl http://localhost:1337/body/test --data ''
```

You get:
```
undefined
error: Sending 500 ("Server Error") response: 
 TypeError: Cannot read property 'shouldFail' of undefined
    at Object.module.exports.test (/home/lobo/dev/luislobo/sails-body-error/api/controllers/BodyController.js:11:29)
    at wrapper (/home/lobo/dev/luislobo/sails-body-error/node_modules/sails/node_modules/lodash/index.js:3095:19)
    at routeTargetFnWrapper (/home/lobo/dev/luislobo/sails-body-error/node_modules/sails/lib/router/bind.js:181:5)
    at callbacks (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:164:37)
    at param (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:138:11)
    at param (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:135:11)
    at pass (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:145:5)
    at nextRoute (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:100:7)
    at callbacks (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:167:11)
    at alwaysAllow (/home/lobo/dev/luislobo/sails-body-error/node_modules/sails/lib/hooks/policies/index.js:224:11)
    at routeTargetFnWrapper (/home/lobo/dev/luislobo/sails-body-error/node_modules/sails/lib/router/bind.js:181:5)
    at callbacks (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:164:37)
    at param (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:138:11)
    at param (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:135:11)
    at pass (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:145:5)
    at nextRoute (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:100:7)
    at callbacks (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:167:11)
    at module.exports (/home/lobo/dev/luislobo/sails-body-error/node_modules/sails/lib/hooks/cors/clear-headers.js:14:3)
    at routeTargetFnWrapper (/home/lobo/dev/luislobo/sails-body-error/node_modules/sails/lib/router/bind.js:181:5)
    at callbacks (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:164:37)
    at param (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:138:11)
    at pass (/home/lobo/dev/luislobo/sails-body-error/node_modules/express/lib/router/index.js:145:5)
```

Stop Sails.

Now do:
```bash
npm install sails@0.12.3
```

Start sails

Curl to the same endpoint

Now you get:
```
{}
undefined
```

**Comment by luislobo** [comment]
This issue kind of break tons of our unit tests.

**Comment by mikermcneil** [comment]
@luislobo thanks for the detailed info!  Looking into it (probably related to the performance patch in Skipper: https://github.com/balderdashy/skipper/commit/54a45f42a21319fea31b89c9413ad1136791a0d7)

**Comment by mikermcneil** [comment]
@luislobo ok, I believe that's it.  @sgress454 is about to publish a new patch release of Skipper that resolves this  (when skipping the body parser as an optimization, it also `req.body = {}` for compatibility).

**Comment by sgress454** [comment]
Ok, all set.  Published as 0.6.4 and added a test for empty-body handling.  Thanks @luislobo!

**Comment by luislobo** [comment]
For the ones still having this issue, you can just run 'rm -rf node_modules && npm install' and it should update your dependencies with the fixed module.
